numpy/_core/multiarray.pyi

```diff
@@ -188,6 +196,17 @@ def array(
     like: None | _SupportsArrayFunc = ...,
 ) -> _ArrayType: ...
```

Comment:
Because the bounds of `_ArrayType` are now changed so that they also include a potential shape type, the `ndmin` parameter is now able to change the return type when `ndmin > 0`.